pkg/store/mock/scanner.go

```diff
@@ -0,0 +1,191 @@
+		} else {
+			m.scanned++
+		}
+	}
+
+	return nil
+}
+
+func convertAssign(dst, src any) error {
+	switch s := src.(type) {
+	case string:
+		switch d := dst.(type) {
+		case *string:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = s
+			return nil
+		case *[]byte:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = []byte(s)
+			return nil
+		}
+	case []byte:
+		switch d := dst.(type) {
+		case *string:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = string(s)
+			return nil
+		case *any:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = bytes.Clone(s)
+			return nil
+		case *[]byte:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = bytes.Clone(s)
+			return nil
+		}
+	case time.Time:
+		switch d := dst.(type) {
+		case *time.Time:
+			*d = s
+			return nil
+		case *string:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = s.Format(time.RFC3339Nano)
+			return nil
+		}
+	case nil:
+		switch d := dst.(type) {
+		case *any:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = nil
+			return nil
+		case *[]byte:
+			if d == nil {
+				return ErrNilPtr
+			}
+			*d = nil
+			return nil
+		}
+	}
+
+	// Could not conver the type src or dst unrecognized.
+	return ErrUnknownType
+}
+
+// Assert that the expected number of `data` items were scanned successfully.
+func (m *MockScanner) AssertScanned(t testing.TB, expected int) {
+	m.Log(t)
+	require.Equal(t, expected, m.scanned, "expected %d scans, got %d", expected, m.scanned)
+}
+
+// Assert that the expected number of `data` items were *not* scanned (ignored/nil).
+func (m *MockScanner) AssertNotScanned(t testing.TB, expected int) {
+	m.Log(t)
+	require.Equal(t, expected, m.notScanned, "expected %d non-scans, got %d", expected, m.notScanned)
+}
+
+// Sets an error to be returned from the scanner when `Scan()` is called. SetError
+// will panic if `data` is already set.
+func (m *MockScanner) SetError(err error) {
+	// we probably don't want to set both of these at the same
+	if m.data != nil {
+		panic("data is not nil so data would not be returned")
+	}
+	m.err = err
+}
+
+// Sets data to be scanned into the destinations given when `Scan()` is called.
+// SetData will panic if `err` is already set.
+func (m *MockScanner) SetData(data []any) {
+	// we probably don't want to set both of these at the same
+	if m.err != nil {
+		panic("err is not nil so data would not be returned")
+	}
+	m.data = data
+}
+
+// Resets the MockScanner to it's original state.
+func (m *MockScanner) Reset() {
+	m.err = nil
+	m.data = nil
+	m.scanned = 0
+	m.notScanned = 0
+	m.testLog = nil
+}
+
+// Logs any test log messages that were recorded during the test; then clears the log.
+func (m *MockScanner) Log(t testing.TB) {
+	for _, log := range m.testLog {
+		t.Log(log)
+	}
+	m.testLog = nil
+}
+
+// Internal method to append a log message to the test log.
+func (m *MockScanner) logf(format string, args ...any) {
+	m.testLog = append(m.testLog, fmt.Sprintf(format, args...))
+}
```

Comment:
I liked your Print statement for debugging; but I wanted to add it to the testing log instead so we could handle it more gracefully; so I added these functions to help out with test logging.